File: presentation/src/test/java/io/tripled/social/client/presentation/controller/FollowControllerTest.java
package io.tripled.social.client.presentation.controller;

import io.tripled.social.client.test.SpiedFollowUserUseCase;
import io.tripled.social.client.presentation.TestOutput;
import io.tripled.social.client.presentation.request.FollowCliRequest;
import org.junit.Before;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.assertThat;

public class FollowControllerTest {

  private FollowController followController;
  private SpiedFollowUserUseCase service;

  @Before
  public void setUp() {
    service = new SpiedFollowUserUseCase();
    followController = new FollowController(service);
  }

  @Test
  public void executesCommand() {
    FollowCliRequest request = new FollowCliRequest("Charlie", "Alice");

    followController.execute(request, new TestOutput());

    assertThat(service.userName, equalTo("Charlie"));
    assertThat(service.userNameToFollow, equalTo("Alice"));
  }

}

File: src/test/kotlin/io/tripled/social/client/infrastructure/cli/controller/FollowControllerTest.kt
package io.tripled.social.client.infrastructure.cli.controller

import io.tripled.social.client.application.usecase.spies.FollowUserSpy
import io.tripled.social.client.infrastructure.cli.TestOutput
import io.tripled.social.client.infrastructure.cli.request.FollowCliRequest
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test

internal class FollowControllerTest {
    private val service: FollowUserSpy = FollowUserSpy()
    private val followController: FollowController = FollowController(service)

    @Test
    fun executesCommand() {
        val request = FollowCliRequest("Charlie", "Alice")

        followController!!.execute(request, TestOutput())

        MatcherAssert.assertThat<String?>(service!!.userName, Matchers.equalTo<String?>("Charlie"))
        MatcherAssert.assertThat<String?>(service!!.userNameToFollow, Matchers.equalTo<String?>("Alice"))
    }
}
